docs/applications/automate_airflow_varsconns/index.md

```diff
@@ -0,0 +1,194 @@
+> You have successfully made the case for migrating ad-hoc workflow scripts over to an orchestration tool like Airflow. 
+> You want to write workflows that break much less often and when they do break, you want them to be easy to fix.
+> You have been given a cloud server, a Linux machine instance, to begin migrating existing scripts over to Airflow. 
+> You were able to install and run a basic DAG. Where do you start?
+
+I recommend starting with Variables and Connections. 
+They form the building blocks of your DAGs because they define your sources, staging areas, and destinations.
+Automating Variables and Connections is just another piece of transparent and reproducible workflows.
+
+## Leveraging Airflow’s Command Line Interface [(CLI)](https://airflow.apache.org/docs/stable/cli.html)
+You can manage your Variables and Connections in Airflow’s webserver with manual input; clicks and typing. 
+However, this work is not efficient and makes you a bottleneck. 
+I recommend leveraging Airflow’s Command Line Interface (CLI) built-in commands.
+
+There are built in airflow command line interface commands that allow you to set, edit, and delete airflow objects like variables and connections.
+
+### Variables CLI
+Store your variables in a key-value structure with a JSON document. An example JSON document would look like this:
+```json
+{
+       "AWS_ACCESS_KEY_ID": "AAAAAAAAAAAA",
+       "AWS_SECRET_ACCESS_KEY": "BBBBBBBBBBBB",
+       "mysql_db": "dbname",
+       "relationaldb_host": "your_database.us-weast-1.rds.amazonaws.com",
+       "relationaldb_port": "9999",
+       "relationaldb_pw": "airflow",
+       "relationaldb_user": "airflow",
+       "s3_bucket": "data_staging",
+       "google_analytics": "[\"website_name\", \"view_id\"]",
+       "TEAM_NAMES": "[\"name_1\",\"name_2\",\"name_3\",\"name_n\"]"
+}
+```
+
+In order to load all the variables above, you only need one terminal command:
+```bash
+airflow variables --import /home/your_user/airflow/your_vars.json
+```
+Save this command for later.
+
+### Connections CLI
+Airflow Connections are currently handled a bit differently than Variables. 
+You cannot store in a JSON document and load them with a single command. 
+I recommend using a bash script.
+
+Use an executable bash script. 
+I have provided an example bash script with three types of connections: 
+    1. AWS IAM Credentials stored in the Connection’s “extra” field 
+    2. Multi-Line JSON API Credentials stored in the Connection’s “extra” field
+    3. An Example Database Connection
+
+There is a more detailed explanation after this example script.
+```bash
+#!/usr/bin/env bash
+ 
+airflow connections -d --conn_id aws_iam_user
+airflow connections -d --conn_id googleanalytics_rest_api
+airflow connections -d --conn_id db_conn
+ 
+airflow connections -a --conn_id aws_iam_user --conn_type s3 --conn_extra '{"aws_access_key_id":"AAAAAAAAAAAAAAA","aws_secret_access_key": "BBBBBBBBBBBB"}'
```

Comment:
Hello @ocelotdata. I am editing your submission and I had a general question. It is not clear to me the use of the example in the "Variables CLI" section, when you have to provide some of the connection details again in the script? Are you not able to access those variables to pass them to your connection script? I see you state that passing a JSON file to the CLI commands won't work. Would you mind clarifying what is the purpose of creating the variables in the first section? Do they provide configuration variables to be used differently later with Airflow? Thanks for these details!